pkg/cri/sbserver/image_pull.go

```diff
@@ -93,9 +95,12 @@ import (
 
 // PullImage pulls an image with authentication config.
 func (c *criService) PullImage(ctx context.Context, r *runtime.PullImageRequest) (*runtime.PullImageResponse, error) {
+	ctx, span := tracing.StartSpan(ctx, makeSpanName("PullImage"))
+	defer span.End()
 	imageRef := r.GetImage().GetImage()
 	namedRef, err := distribution.ParseDockerRef(imageRef)
 	if err != nil {
+		span.RecordError(err)
```

Comment:
For RecordError calls, how about renaming this function to something like pullImage and calling it like below? Calling RecordError() on all branches seems error-prone.

```
resp, err := pullImage(...)
if err != nil {
  span.RecordError(err)
```

Reply:
Good point. I can take a look into creating a thin wrapper

Reply:
I'm also concerned about the pattern of `span.RecordError(err)` in each branch; that's not going to be easy to enforce as we continue to maintain the code.

I'd prefer the wrapper, but another reasonable approach would be to change the signature to use a named return:

```go
func (c *criService) PullImage(ctx context.Context, r *runtime.PullImageRequest) (_ *runtime.PullImageResponse, err error)
```

and then capture the error in a `defer`:

```go
defer func(){
	if err != nil {
		span.RecordError(err)
	}
}()
```

On further reflection: for `sbserver` and `server`, can we create the spans inside a gRPC interceptor and stuff them into ctx if we later want to add attributes?

Reply:
Hi @samuelkarp, I already updated this PR to add a wrapper and removed most of the RecordError calls. Please take a look. I am fine changing the current impl to use a defer func with named error, if we strongly feel about it. 

> for sbserver and server, can we create the spans inside a gRPC interceptor and stuff them into ctx if we later want to add attributes

IIUC we already have a span from the gRPC interceptor which is currently the top level span. This PR adds additional child spans for tracing after the gRPC calls is received. Sure, we can keep using the parent gRPC span as well to add the attributes, but I think ability to trace which packages are being invoked is an added advantage.